<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [rabbitmq-discuss] AlreadyClosedException always sets hardError	and initiatedByApplication to true
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20AlreadyClosedException%20always%20sets%20hardError%0A%09and%20initiatedByApplication%20to%20true&In-Reply-To=%3C5BFF87E3-BBD7-421D-8CD2-C8AE6E8A7FD8%40rabbitmq.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="031017.html">
   <LINK REL="Next"  HREF="031019.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[rabbitmq-discuss] AlreadyClosedException always sets hardError	and initiatedByApplication to true</H1>
    <B>Michael Klishin</B> 
    <A HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20AlreadyClosedException%20always%20sets%20hardError%0A%09and%20initiatedByApplication%20to%20true&In-Reply-To=%3C5BFF87E3-BBD7-421D-8CD2-C8AE6E8A7FD8%40rabbitmq.com%3E"
       TITLE="[rabbitmq-discuss] AlreadyClosedException always sets hardError	and initiatedByApplication to true">mklishin at gopivotal.com
       </A><BR>
    <I>Tue Oct 15 23:45:32 BST 2013</I>
    <P><UL>
        <LI>Previous message: <A HREF="031017.html">[rabbitmq-discuss] AlreadyClosedException always sets hardError and initiatedByApplication to true
</A></li>
        <LI>Next message: <A HREF="031019.html">[rabbitmq-discuss] AlreadyClosedException always sets hardError and initiatedByApplication to true
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#31018">[ date ]</a>
              <a href="thread.html#31018">[ thread ]</a>
              <a href="subject.html#31018">[ subject ]</a>
              <a href="author.html#31018">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>On oct 16, 2013, at 1:24 a.m., Jonathan Halterman &lt;<A HREF="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">jhalterman at gmail.com</A>&gt; wrote:

&gt;<i> I noticed that AlreadyClosedException always sets hardError and initiatedByApplication to true, even if the error was not hard (only the channel was closed) and the channel was not closed by the application. This makes it more difficult to properly recover a closed resource if I can't easily determine which resource was closed and why. Could this be fixed?
</I>
Jonathan,

This indeed sounds like a bug.

How can we reproduce it? Simply attempt an operation on a closed channel?
Does it matter how exactly the channel was closed?

MK



-------------- next part --------------
A non-text attachment was scrubbed...
Name: signature.asc
Type: application/pgp-signature
Size: 495 bytes
Desc: Message signed with OpenPGP using GPGMail
URL: &lt;<A HREF="http://lists.rabbitmq.com/pipermail/rabbitmq-discuss/attachments/20131016/437ec59b/attachment.pgp">http://lists.rabbitmq.com/pipermail/rabbitmq-discuss/attachments/20131016/437ec59b/attachment.pgp</A>&gt;
</PRE>
















































































<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="031017.html">[rabbitmq-discuss] AlreadyClosedException always sets hardError and initiatedByApplication to true
</A></li>
	<LI>Next message: <A HREF="031019.html">[rabbitmq-discuss] AlreadyClosedException always sets hardError and initiatedByApplication to true
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#31018">[ date ]</a>
              <a href="thread.html#31018">[ thread ]</a>
              <a href="subject.html#31018">[ subject ]</a>
              <a href="author.html#31018">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">More information about the rabbitmq-discuss
mailing list</a><br>
</body></html>
